packages/mapviewer/src/utils/components/DropdownButton.vue

```diff
@@ -1,88 +1,134 @@
 /**
- * Manages a Bootstrap dropdown.
+ * Component to render a dropdown button.
  *
  * It is possible to have the dropdown attached to an extra button with a caret, instead of all
- * inline. For that you have to add the with-toggle-button HTML attribute to the component use
+ * inline. For that you have to add the with-toggle-button props to the component use.
  */
-
+import { size, useFloating } from '@floating-ui/vue'
 import { randomIntBetween } from '@geoadmin/numbers'
-import { Dropdown } from 'bootstrap'
-import { onBeforeUnmount, onMounted, ref, useTemplateRef, watch } from 'vue'
+import GeoadminTooltip from '@geoadmin/tooltip'
+import { onClickOutside } from '@vueuse/core'
+import { computed, ref, useTemplateRef, watch } from 'vue'
 import { useI18n } from 'vue-i18n'
 
-const { title, withToggleButton, disabled, small } = defineProps({
-    title: {
-        type: String,
-        required: true,
-    },
-    withToggleButton: {
-        type: Boolean,
-        default: false,
-    },
-    disabled: {
-        type: Boolean,
-        default: false,
-    },
-    small: {
-        type: Boolean,
-        default: false,
-    },
-})
+export interface DropdownItem {
+    id: string | number
+    title: string
+    value: string | number | object
+    description?: string
+}
 
-const emits = defineEmits({ click: () => true, selectItem: (item) => item?.id && item?.title })
+const {
+    title,
+    items,
+    currentValue,
+    withToggleButton = false,
+    disabled = false,
+    small = false,
+} = defineProps<{
+    title: string
+    items: DropdownItem[]
+    currentValue: string | number | object
+    withToggleButton?: boolean
+    disabled?: boolean
+    small?: boolean
+}>()
 
-const { t } = useI18n()
+const emits = defineEmits<{
+    click: []
+    selectItem: [item: DropdownItem]
+}>()
 
-const dropdownMenu = useTemplateRef('dropdownMenu')
-const dropdownToggleButton = useTemplateRef('dropdownToggleButton')
-const dropdownMainButton = useTemplateRef('dropdownMainButton')
+const isOpen = ref<boolean>(false)
 
-// generating a unique HTML ID for this dropdown
-const uniqueHtmlId = ref(`dropdown-${randomIntBetween(0, 10000)}`)
+const { t } = useI18n()
+
+const dropdownMenuRef = useTemplateRef<HTMLUListElement>('dropdownMenu')
+const dropdownToggleButtonRef = useTemplateRef<HTMLButtonElement>('dropdownToggleButton')
+const dropdownMainButtonRef = useTemplateRef<HTMLButtonElement>('dropdownMainButton')
 
-let dropdown = null
-onMounted(() => {
+const anchorElement = computed<HTMLElement | null>(() => {
     if (withToggleButton) {
-        dropdown = new Dropdown(dropdownToggleButton.value)
-    } else {
-        dropdown = new Dropdown(dropdownMainButton.value)
+        return dropdownToggleButtonRef.value
     }
+    return dropdownMainButtonRef.value
 })
-onBeforeUnmount(() => {
-    dropdown?.dispose()
+
+// generating a unique HTML ID for this dropdown
+const uniqueHtmlId = ref<string>(`dropdown-${randomIntBetween(0, 10000)}`)
```

Comment:
There is a very small chance that this can have collisions, but I think the interval is big enough so that it's not a common occurrence :)

Reply:
akshually (:wink: ) I think we have the uuid package installed in mapviewer?

Reply:
you're right, that could be a good solution to simply use UUID... let's do that (I raised the "seed) to 100'000 but UUID will still be better as it will be based on the timestamp anyway...)